src/suggestions/index.jsx

```diff
@@ -235,70 +235,48 @@ export const SuggestionsMixin = {
   },
 
   ensureSelectedSuggestionVisibility() {
-    if (!this.suggestionsMixin_suggestionNodes) {
+    if (this.suggestionsAbove || !this.suggestionsMixin_suggestionNodes) {
       return;
     }
 
-    const suggestionElement = ReactDOM.findDOMNode(
-      this.suggestionsMixin_suggestionNodes[this.state.selectedSuggestionId]
-    );
+    const suggestionElement = this.suggestionsMixin_suggestionNodes[
+      this.state.selectedSuggestionId
+    ];
 
     if (!suggestionElement) {
       return;
     }
 
-    const offsetTop = getOffsetTop(suggestionElement);
+    const offsetTop = getOffsetTop(suggestionElement) + suggestionElement.offsetHeight;
 
-    if (offsetTop - window.pageYOffset > 0) {
+    if (offsetTop > window.innerHeight) {
       suggestionElement.scrollIntoView();
     }
-
-    if (window.pageYOffset + window.innerHeight <= offsetTop) {
-      suggestionElement.scrollIntoView();
-    }
-  },
-
-  suggestionsMixin_storeSuggestionNode(ref) {
-    if (!ref) {
-      return;
-    }
-
-    this.suggestionsMixin_suggestionNodes = {
-      ...this.suggestionsMixin_suggestionNodes,
-      [ref.props['data-suggestion-id']]: ref,
-    };
   },
 
   renderSuggestions() {
-    if (!this.state.suggestionsVisible) {
-      return;
-    }
-
-    const query = escapeRegExp(this.state.suggestionsQuery);
-    const matcher = new RegExp(`^${query}| ${query}`, 'i'); // start of string, or preceded by a space
+    const { suggestions, suggestionsVisible, selectedSuggestionId } = this.state;
 
-    const suggestions = this.props.suggestions
-      .filter(({ name }) => matcher.test(name))
-      .slice(0, 10);
-
-    if (!suggestions.length) {
-      return null;
+    if (!suggestionsVisible || !suggestions.length) {
+      return;
     }
 
-    const selectedSuggestionId = this.state.selectedSuggestionId || suggestions[0].ID;
-
     return (
       <div
         className="wpnc__suggestions"
+        ref={div => (this.suggestionsMixin_suggestionList = div)}
         onMouseEnter={() => (this.suggestionsCancelBlur = true)}
         onMouseLeave={() => (this.suggestionsCancelBlur = false)}
       >
         <ul>
           {suggestions.map(suggestion =>
             <Suggestion
-              ref={this.suggestionsMixin_storeSuggestionNode}
-              data-suggestion-id={suggestion.ID}
               key={'user-suggestion-' + suggestion.ID}
+              getElement={suggestionElement =>
+                (this.suggestionsMixin_suggestionNodes = {
+                  ...this.suggestionsMixin_suggestionNodes,
+                  [suggestion.ID]: suggestionElement,
+                })}
```

Comment:
same note here - do we have a good reason to move this into the JSX vs adding another method to the mixin?